src/main/webapp/resources/js/contexts/AnalysisContext.js

```diff
@@ -51,12 +55,30 @@ const AnalysisContext = React.createContext(initialContext);
 
 function AnalysisProvider(props) {
   const [analysisContext, setAnalysisContext] = useState(initialContext);
+  const analysisIdentifier = window.location.pathname.match(/analysis\/(\d+)/)[1];
```

Comment:
Move this into you loading useEffect that way it does not happen on every re-render.  You can store the value in a setState variable by the same name.

Reply:
Updated in 91b5b8f9ea189503d79174656273a9f018de1d60